src/test/java/com/teragrep/pth10/TeragrepTransformationTest.java

```diff
@@ -791,4 +794,93 @@ public void tgHdfsSaveAfterBloomEstimateTestUsingRegexExtract() {
                     Assertions.assertEquals(Collections.singletonList("5"), listOfResult);
                 });
     }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgSetConfigStringTest() {
+        Config fakeConfig = ConfigFactory
+                .defaultApplication()
+                .withValue("dpl.pth_00.dummy.value", ConfigValueFactory.fromAnyRef("oldValue"));
+        streamingTestUtil.getCtx().setConfig(fakeConfig);
+        Assertions.assertEquals("oldValue", streamingTestUtil.getCtx().getConfig().getString("dpl.pth_00.dummy.value"));
+        streamingTestUtil
+                .performDPLTest("index=index_A | teragrep set config dpl.pth_00.dummy.value newValue", testFile, ds -> {
+                    Assertions
+                            .assertEquals(
+                                    "newValue",
+                                    streamingTestUtil.getCtx().getConfig().getString("dpl.pth_00.dummy.value")
+                            );
+                });
+    }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgSetConfigLongTest() {
+        Config fakeConfig = ConfigFactory
+                .defaultApplication()
+                .withValue("dpl.pth_00.dummy.value", ConfigValueFactory.fromAnyRef(12345));
+        streamingTestUtil.getCtx().setConfig(fakeConfig);
+        Assertions.assertEquals(12345L, streamingTestUtil.getCtx().getConfig().getLong("dpl.pth_00.dummy.value"));
+        streamingTestUtil
+                .performDPLTest("index=index_A | teragrep set config dpl.pth_00.dummy.value 99999", testFile, ds -> {
+                    Assertions
+                            .assertEquals(
+                                    99999L, streamingTestUtil.getCtx().getConfig().getLong("dpl.pth_00.dummy.value")
+                            );
+                });
+    }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void tgSetConfigMismatchedTypesTest() {
+        Config fakeConfig = ConfigFactory
+                .defaultApplication()
+                .withValue("dpl.pth_00.dummy.value", ConfigValueFactory.fromAnyRef(12345));
+        streamingTestUtil.getCtx().setConfig(fakeConfig);
+        Assertions.assertEquals(12345L, streamingTestUtil.getCtx().getConfig().getLong("dpl.pth_00.dummy.value"));
+        Throwable t = streamingTestUtil
+                .performThrowingDPLTest(
+                        NumberFormatException.class,
+                        "index=index_A | teragrep set config dpl.pth_00.dummy.value stringValue", testFile, ds -> {
+                        }
+                );
+
+        Assertions.assertEquals("For input string: \"stringValue\"", t.getMessage());
```

Comment:
Just wondering from where does this exception message come from as I can't find it, test looks good though.

Reply:
it comes from `Double.parseDouble()`